codetransformer/instructions.py

```diff
@@ -123,7 +122,45 @@ def __repr__(self):
     __str__ = __repr__
 
 
-class Instruction(InstructionMeta._marker, metaclass=InstructionMeta):
+class JumpTarget:
+    """Base class for objects that can be targets of jump instructions.
+
+    This is the base for both Instruction and Label.
+    """
+
+    def __init__(self):
+        self._target_of = set()
+        self._stolen_by = None  # used for lnotab recalculation
+
+    def steal(self, instr):
+        """Steal the jump index off of `instr`.
+
+        This makes anything that would have jumped to `instr` jump to
+        this Instruction instead.
+
+        Parameters
+        ----------
+        instr : JumpTarget
+            The target to steal the jump sources from.
+
+        Returns
+        -------
+        self : JumpTarget
+            The object that owns this method.
+
+        Notes
+        -----
+        This mutates self and ``instr`` inplace.
+        """
+        instr._stolen_by = self
+        for jmp in instr._target_of:
+            jmp.arg = self
+        self._target_of = instr._target_of
```

Comment:
This needs to be an `|=`, but that can go in a separate PR.

Reply:
let's remember to do this